eng/scripts/Update-Mgmt-CI.ps1

```diff
@@ -9,38 +9,41 @@ function Update-CIFile() {
 
     $shouldRemove = $false
 
-    $lines = Get-Content $mgmtCiFile
-    $newLines = [System.Collections.ArrayList]::new()
-    foreach($line in $lines) {
-        if($line.StartsWith("    ArtifactName:")) {
-            Continue
-        }
-        if($line.StartsWith("trigger:") -or $line.StartsWith("pr:")) {
-            $prefix = $line.Substring(0, $line.IndexOf(":") + 1)
-            $newLines.Add("$prefix none") | Out-Null
-            $shouldRemove = $true
-            Continue
-        }
-        if($line.StartsWith("    - name:")) {
-            if($line -match "Azure.ResourceManager.") {
-                $shouldRemove = $false
-            }
-            else {
-                $shouldRemove = $true
-                Continue
-            }
-        }
-        if($shouldRemove) {
-            if($line.StartsWith(" ")) {
-                Continue
-            }
-            $shouldRemove = $false
-        }
+    $content = Get-Content $mgmtCiFile -Raw
 
-        $newLines.Add($line) | Out-Null
+    if ($content -match "(?s)ServiceDirectory: (?<sd>[^\r\n]+).*- name: (?<p>[^\r\n]+)")
+    {
+        $serviceDirectory = $matches["sd"]
+        $packageName = $matches["p"]
+    }
+    else {
+        Write-Error "Could not parse out the service directory and package from $mgmtCiFile, so skipping."
+        return
+    }
+
+    $prtriggers = @"
+pr:
+  branches:
+    include:
+    - main
+    - feature/*
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/$serviceDirectory/$packageName
```

Comment:
should we add another trigger here for `sdk/$serviceDirectory/ci.mgmt.yml` so that the ci runs if someone changes its definition?

Reply:
Fair I can include that.